src/docquery/ext/pipeline_image_classification.py

```diff
@@ -0,0 +1,200 @@
+    add_end_docstrings,
+    is_tf_available,
+    is_torch_available,
+    is_vision_available,
+    logging,
+    requires_backends,
+)
+
+
+if is_vision_available():
+    from PIL import Image
+    from transformers.image_utils import load_image
+
+if is_tf_available():
+    import tensorflow as tf
+    from transformers.models.auto.modeling_tf_auto import TF_MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING
+    from transformers.tf_utils import stable_softmax
+
+if is_torch_available():
+    from transformers.models.auto.modeling_auto import MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING
+
+logger = logging.get_logger(__name__)
+
+
+class ModelType(ExplicitEnum):
+    Standard = "standard"
+    VisionEncoderDecoder = "vision_encoder_decoder"
+
+
+def donut_token2json(tokenizer, tokens, is_inner_value=False):
+    """
+    Convert a (generated) token sequence into an ordered JSON format.
+    """
+    output = dict()
+
+    while tokens:
+        start_token = re.search(r"<s_(.*?)>", tokens, re.IGNORECASE)
+        if start_token is None:
+            break
+        key = start_token.group(1)
+        end_token = re.search(rf"</s_{key}>", tokens, re.IGNORECASE)
+        start_token = start_token.group()
+        if end_token is None:
+            tokens = tokens.replace(start_token, "")
+        else:
+            end_token = end_token.group()
+            start_token_escaped = re.escape(start_token)
+            end_token_escaped = re.escape(end_token)
+            content = re.search(f"{start_token_escaped}(.*?){end_token_escaped}", tokens, re.IGNORECASE)
+            if content is not None:
+                content = content.group(1).strip()
+                if r"<s_" in content and r"</s_" in content:  # non-leaf node
+                    value = donut_token2json(tokenizer, content, is_inner_value=True)
+                    if value:
+                        if len(value) == 1:
+                            value = value[0]
+                        output[key] = value
+                else:  # leaf nodes
+                    output[key] = []
+                    for leaf in content.split(r"<sep/>"):
+                        leaf = leaf.strip()
+                        if leaf in tokenizer.get_added_vocab() and leaf[0] == "<" and leaf[-2:] == "/>":
+                            leaf = leaf[1:-2]  # for categorical special tokens
+                        output[key].append(leaf)
+                    if len(output[key]) == 1:
+                        output[key] = output[key][0]
+
+            tokens = tokens[tokens.find(end_token) + len(end_token) :].strip()
+            if tokens[:6] == r"<sep/>":  # non-leaf nodes
+                return [output] + donut_token2json(tokenizer, tokens[6:], is_inner_value=True)
+
+    if len(output):
+        return [output] if is_inner_value else output
+    else:
+        return [] if is_inner_value else {"text_sequence": tokens}
+
+
+@add_end_docstrings(PIPELINE_INIT_ARGS)
+class ImageClassificationPipeline(Pipeline):
```

Comment:
Since most of this code is duplicated from the transformers library, I think this class should instead subclass the transformers version of this class and then call `super().method_name(...)` wherever we have duplicated code, so that it's clear which pieces of code are new additions that we've added.

Reply:
commented separately about why we copy/pasted.